### ISSUE TITLE
[Bug]: flutter unit test of basic config fails with dynamic library error

### ISSUE THREAD

**Issue by mnimer** [realm/realm-dart#1071]
### What happened?

In the default flutter todo template app, I can not write and run a simple unit test on an intel Mac laptop. I get this error "Failed to load dynamic library 'librealm_dart.dylib'".

The example app will run just fine on mac desktop. It's only the unit test that fails.

### Repro steps

Create a new unit test file (code below) and run debug, or flutter test ...., on the file in IntelliJ or command line. 

### Version

 flutter:  3.7.0-10.0.pre.34  & realm: ^0.6.0+beta

### What Realm SDK flavor are you using?

Local Database only

### What type of application is this?

Flutter Application

### Client OS and version

macos, test cli

### Code snippets

import 'package:flutter_test/flutter_test.dart';
import 'package:flutter_todo/realm/schemas.dart';
import 'package:realm/realm.dart';

void main() {

  test("create db", () async {
    var config = Configuration.local([Item.schema]);
    var realm = Realm(config);

    expect(realm, isNotNull);
  });
}

### Stacktrace of the exception/crash you're getting

```shell
package:realm/src/init.dart 81:3                _openRealmLib
package:realm/src/init.dart 105:24              initRealm
package:realm/src/native/realm_core.dart 70:17  new _RealmCore._
package:realm/src/native/realm_core.dart 78:37  new _RealmCore
package:realm/src/native/realm_core.dart 51:30  realmCore
package:realm/src/native/realm_core.dart        realmCore
package:realm/src/configuration.dart 77:14      Configuration.defaultStoragePath
package:realm/src/configuration.dart 87:47      Configuration.defaultRealmPath
package:realm/src/configuration.dart            Configuration.defaultRealmPath
package:realm/src/configuration.dart 100:44     Configuration._defaultPath
package:realm/src/configuration.dart 96:50      new Configuration._
package:realm/src/configuration.dart 226:14     new LocalConfiguration._
package:realm/src/configuration.dart 137:26     Configuration.local
test/realm_local_test.dart 14:32                main.<fn>

Invalid argument(s): Failed to load dynamic library 'librealm_dart.dylib': dlopen(librealm_dart.dylib, 0x0001): tried: 'librealm_dart.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibrealm_dart.dylib' (no such file), '/Users/mikenimer/Development/libraries/flutter/bin/cache/artifacts/engine/darwin-x64/./librealm_dart.dylib' (no such file), '/Users/mikenimer/Development/libraries/flutter/bin/cache/artifacts/engine/darwin-x64/../../../librealm_dart.dylib' (no such file), '/Users/mikenimer/Development/libraries/flutter/bin/cache/artifacts/engine/darwin-x64/Frameworks/librealm_dart.dylib' (no such file), '/Users/mikenimer/Development/libraries/flutter/bin/cache/artifacts/engine/darwin-x64/./librealm_dart.dylib' (no such file), '/Users/mikenimer/Development/libraries/flutter/bin/cache/artifacts/engine/darwin-x64/../../../librealm_dart.dylib' (no such file), '/Users/mikenimer/Development/libraries/flutter/bin/cache/artifacts/engine/darwin-x64/Frameworks/librealm_dart.dylib' (no such file), '/usr/lib/librealm_dart.dylib' (no such file, not in dyld cache), 'librealm_dart.dylib' (no such file), '/usr/local/lib/librealm_dart.dylib' (no such file), '/usr/lib/librealm_dart.dylib' (no such file, not in dyld cache)
```

**Comment by dotjon0** [comment]
@mnimer may be worth trying the latest stable release of flutter 3.3.10 and the latest version of realm-dart 0.8.0+rc - do you get the same with this?

**Comment by mnimer** [comment]
> @mnimer may be worth trying the latest stable release of flutter 3.3.10 and the latest version of realm-dart 0.8.0+rc - do you get the same with this?

I did try upgrading to the latest (3.3.10 & 0.8.0+rc) and I'm getting the same error

**Comment by desistefanova** [comment]
This issue will be fixed after landing this PR https://github.com/realm/realm-dart/pull/929

**Comment by desistefanova** [comment]
The new release is available here [realm 0.10.0+rc](https://pub.dev/packages/realm).
You can upgrade the realm to `0.10.0+rc` and then to run the new available command `flutter pub run realm install` from the project root folder. This will download the library and you will be able to run your tests.
I'm closing this issue. Feel free to reopen it in case of any related problems.
